flink-container/src/main/java/org/apache/flink/container/entrypoint/StandaloneJobClusterConfigurationParserFactory.java

```diff
@@ -79,6 +92,20 @@ public StandaloneJobClusterConfiguration createResult(@Nonnull CommandLine comma
 			hostname,
 			restPort,
 			jobClassName,
-			savepointRestoreSettings);
+			savepointRestoreSettings,
+			jobId);
+	}
+
+	private static JobID getJobId(CommandLine commandLine) {
+		String jobId = commandLine.getOptionValue(JOB_ID_OPTION.getOpt());
+		if (jobId == null) {
+			return DEFAULT_JOB_ID;
+		}
+
+		try {
+			return JobID.fromHexString(jobId);
+		} catch (Throwable t) {
+			throw new IllegalArgumentException("Failed to parse job ID '" + jobId + "'", t);
```

Comment:
add a description of what format is expected

Reply:
Good idea. Is the updated message OK?